__tests__/integrations/BingAds/browser.test.js

```diff
@@ -0,0 +1,148 @@
+import { BingAds } from '../../../src/integrations/BingAds';
+import { event, query, products } from './__mocks__/data';
+
+describe('BingAds init tests', () => {
+  let Bingads;
+  test('Testing init call of Bing ads', () => {
+    Bingads = new BingAds({ tagID: '12567839' }, { loglevel: 'debug' });
+    Bingads.init();
+    expect(typeof window.bing12567839).toBe('object');
+  });
+});
+
+describe('BingAds page', () => {
+  let bingAds;
+  const output = [];
+  beforeEach(() => {
+    bingAds = new BingAds({ tagID: '12567839' }, { loglevel: 'debug' });
+    bingAds.init();
+    window.bing12567839.push = jest.fn((x) => output.push(x));
+  });
+
+  test('send pageview', () => {
+    bingAds.page({
+      context: {},
+      properties: {
+        category: 'test category',
+        path: '/test',
+        url: 'http://localhost',
+        referrer: '',
+        title: 'test page',
+        testDimension: 'abc',
+      },
+    });
+    expect(output[0]).toEqual('pageLoad');
+  });
+});
+
+describe('BingAds Track event', () => {
+  let bingAds;
+  const output = [];
+  beforeEach(() => {
+    bingAds = new BingAds({ tagID: '12567839' }, { loglevel: 'DEBUG' });
+    bingAds.init();
+    window.bing12567839.push = jest.fn((x, y, z) => output.push({ event: y, ...z }));
```

Comment:
whats the purpose of x?

Reply:
x represent "event" 
`window[this.uniqueId].push('event', eventToSend, payload);`

Reply:
But we are not using the variable name itself i.e x is not used in the function implementation

Reply:
In test cases we need 2nd and 3rd argument to validate. First argument "x" we are not using but we can't remove it.

Reply:
ok